variants/FYSETC_S6/ldscript.ld

```diff
@@ -0,0 +1,187 @@
+/*
+*****************************************************************************
+**
+
+**  File        : LinkerScript.ld
+**
+**  Abstract    : Linker script for STM32F407VETx Device with
+**                512KByte FLASH, 128KByte RAM
+**
+**                Set heap size, stack size and stack location according
+**                to application requirements.
+**
+**                Set memory bank area and size if external memory is used.
+**
+**  Target      : STMicroelectronics STM32
+**
+**
+**  Distribution: The file is distributed as is, without any warranty
+**                of any kind.
+**
+*****************************************************************************
+** @attention
+**
+** <h2><center>&copy; COPYRIGHT(c) 2014 Ac6</center></h2>
+**
+** Redistribution and use in source and binary forms, with or without modification,
+** are permitted provided that the following conditions are met:
+**   1. Redistributions of source code must retain the above copyright notice,
+**      this list of conditions and the following disclaimer.
+**   2. Redistributions in binary form must reproduce the above copyright notice,
+**      this list of conditions and the following disclaimer in the documentation
+**      and/or other materials provided with the distribution.
+**   3. Neither the name of Ac6 nor the names of its contributors
+**      may be used to endorse or promote products derived from this software
+**      without specific prior written permission.
+**
+** THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+** AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+** IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+** DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+** FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+** DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+** SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+** CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+** OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+** OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+**
+*****************************************************************************
+*/
+
+/* Entry Point */
+ENTRY(Reset_Handler)
+
+/* Highest address of the user mode stack */
+_estack = 0x20020000;    /* end of RAM */
+/* Generate a link error if heap and stack don't fit into RAM */
+_Min_Heap_Size = 0x200;;      /* required amount of heap  */
+_Min_Stack_Size = 0x400;; /* required amount of stack */
+
+/* Specify the memory areas */
+MEMORY
+{
+FLASH (rx)      : ORIGIN = 0x8010000, LENGTH = 512K
```

Comment:
This board has bootloader inside. So i change the start address .

Reply:
Bootloader ... then there are several points to clarify:
* LENGTH should reflect the remaining size of Flash not the total.
What could be done is to do that:
```
RAM (xrw)      : ORIGIN = 0x20000000, LENGTH = LD_MAX_DATA_SIZE
FLASH (rx)      : ORIGIN = 0x8000000 + LD_FLASH_OFFSET, LENGTH = LD_MAX_SIZE - LD_FLASH_OFFSET
```
And to define LD_FLASH_OFFSET , in boards.txt you should simply add
`3dprinter.menu.pnum.FYSETC_S6.build.flash_offset=0x10000 `

* Due to bootloader, it is necessary to move Vector table (in RAM or FLASH). But I have not seen some code related to this.
One solution is to mimic what has been done for variant MALYANM300_F070CB, there is an extra flag added to the compiler:
`3dprinter.menu.pnum.MALYANM300_F070CB.build.extra_flags=...  -DVECT_TAB_OFFSET={build.flash_offset}`
It reuses 'flash_offset' that has been define just before, and vector table will be moved to that offset in Flash (after the bootloader).

* What upload method do you intend to use to flash the board ?
By default this variant inherit from 3Dprinter menu, 3 methods based on STM32CubeProgrammer, but currently, in Arduino, it is not  compatible with a bootloader.
Will it be possible to flash directly from Arduino IDE ?
It may depends on bootloader itself. Can you tell us more about the bootloader ?

Reply:
The bootloader update the firmware using SPI sdcard . And if Arduino is not compatible with a bootloader using STM32CubeProgrammer(DFU) mode , i think it's not possible to upload the firmware with it. We can only use arduino to compile the firmware. And S6 can use the STM32CubeProgrammer to upload the firmware , you can see it here : https://wiki.fysetc.com/FYSETC_S6/